Commit to stabilize RHSM operations.  This code is derived from contrib

mazzystr opened the pull request:
Commit to stablize RHSM operations.  This code is derived from contrib.

Tested six deployments in the last 12 hours with no failures with reg, pool attach, yum repo enable.

sdodson commented on playbooks/init/repos.yml:
```diff
@@ -1,8 +1,12 @@
   tasks:
+  - name: subscribe instance to Red Hat Subscription Manager
+    include_role:
+      name: rhel_subscribe
```
It looks like including this role without defining the variables it expects will result in a failure. Only include the role when the required variables are defined?

sdodson commented:
/assign sdodson

mazzystr commented:
/retest

sdodson commented on roles/rhel_subscribe/tasks/main.yml:
```diff
@@ -3,17 +3,23 @@
+    rhel_subscription_pool: "{{ lookup('env', 'rhel_subscription_pool') | default(rhsub_pool | default('Red Hat OpenShift Container Platform, Premium*')) }}"
+    rhel_subscription_user: "{{ lookup('env', 'rhel_subscription_user') | default(rhsub_user | default(omit, True)) }}"
+    rhel_subscription_pass: "{{ lookup('env', 'rhel_subscription_pass') | default(rhsub_pass | default(omit, True)) }}"
+    rhel_subscription_server: "{{ lookup('env', 'rhel_subscription_server') | default(rhsub_server | default(omit, True)) }}"
```
Move these to defaults/main.yml to avoid the set_fact?

sdodson commented on roles/rhel_subscribe/tasks/main.yml:
```diff
@@ -33,36 +39,58 @@
+  command: subscription-manager version
+  register: rh_subscribed
+  changed_when: "'not registered' in rh_subscribed.stdout"
+  ignore_errors: yes
```
`failed_when: false` so that we don't affect playbook return code

mazzystr replied:
Ack

sdodson commented on roles/rhel_subscribe/tasks/main.yml:
```diff
@@ -33,36 +39,58 @@
-  register: subscribe_pool
-  until: subscribe_pool | succeeded
-  when: openshift_pool_id.stdout != ''
+  shell: "subscription-manager attach --pool {{ rhel_subscription_pool }}"
```
Seems like command module is fine here?

sdodson commented on roles/rhel_subscribe/tasks/enterprise.yml:
```diff
@@ -1,18 +1,47 @@
+    - set_fact:
+        default_ose_version: '3.7'
+
+    - set_fact:
```
We're really trying to reduce the use of set_fact, move this to defaults/main.yml ?

mazzystr replied:
Better yet... OpenShift version fail condition should not be here.  All the tasks relating to ose_version can simply be removed.

mazzystr commented:
I think we finally have a good commit here.  

@sdodson I addressed your concerns.  Take notice of role openshift_role now containing a play and set of tasks to register to rhsm yum repos.  If we don't like this we can easily move play rhel_roles back to role rhel_subscribe.

sdodson commented:
Linter failed here, fix this then it lgtm.
```
playbooks/init/repos.yml:

  10:7      error    wrong indentation: expected 4 but found 6  (indentation)
```

sdodson commented:
/lgtm
/test install

mazzystr commented:
@sdodson Thank you sir.  I don't have merge access...can you hit the Merge button?

sdodson commented:
@mazzystr Normally we rely on the bot to merge everything and the flow is that once all tests are green and the PR has lgtm label it's entered into the merge queue. However the jenkins instance just fell over and lost all its jobs, after reviewing the install job it looks like the install was completed successfully and I don't see anyway that this could break the merge queue in the future so I'll merge it.

mazzystr commented:
Thank you!  Pleasure doing business with you! :)
